pwndbg/commands/context.py

```diff
@@ -180,23 +196,52 @@ def context_code():
         if closest_line < 0:
             return []
 
-        source = gdb.execute('list %i' % closest_line, from_tty=False, to_string=True)
+        # Get the full source code
+        filename = symtab.fullname()
+        source = get_highlight_source(filename)
 
         # If it starts on line 1, it's not really using the
         # correct source code.
         if not source or closest_line <= 1:
             return []
 
-        # highlight the current code line
-        source_lines = source.splitlines()
-        if pwndbg.config.highlight_source:
-            for i in range(len(source_lines)):
-                if source_lines[i].startswith('%s\t' % closest_line):
-                    source_lines[i] = C.highlight(source_lines[i])
-                    break
-
-        banner = [pwndbg.ui.banner("source")]
-        banner.extend(source_lines)
+        n = int(source_code_length)
+
+        # Compute the line range
+        st = max(closest_line - 1 - n//2, 0)
+        ed = min(closest_line - 1 + n//2 + 1, len(source))
+        num_width = len(str(ed))
+
+        # split the code
+        source = source[st:ed]
+
+        # Format the output
+        formated_source = []
+        for line_number, code in enumerate(source, start=st + 1):
+            # Compute the prefix
+            # XXX: use nearpc-prefix here, provide customize?
+
+            # XXX: copy from color/disasm
+            prefix_sign = pwndbg.config.nearpc_prefix
+            value = prefix_sign.value
+            if isinstance(value, bytes):
+                value = codecs.decode(value, 'utf-8')
```

Comment:
Is it necessary?

Reply:
Not sure.
In `disasm.py`, there is the same code to compute the Unicode prefix length. I think the reason is that when we use the command line to set the prefix, it will be utf8, not the unicode.
But I didn't verify this.

A better way is that we create some wrapper in `@pwndbg.config.Trigger` to make sure everything in config is unicode.